### PR TITLE
Amended CRM + UI code to allow showing DQT roles for a user

### PR DESCRIPTION
**Pull request by hortha**
### Context

The add and edit user journeys were put together quickly without designs so we could bootstrap an environment. After a review, some improvements were suggested.

### Changes proposed in this pull request

Amend the /users/{userId} page to list the DQT roles the user has (if they have an active CRM account).

Amend the warning messages on /users/{userId} shown when user does not have a CRM account or when it’s disabled by appending ‘The user will not be able to access the TRS Console.’

Amend the /users/add/confirm page to show the same CRM roles list and warning messages as the /users/userId page.

### Guidance to review

Tweak to CRM query + Changes to UI + tests

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [x] Cleaned commit history
-   [x] Tested by running locally

### CODE REVIEW

**Comment by hortha** [comment]
[TRS user management improvements](https://trello.com/c/BSeTkhnW/1621-trs-user-management-improvements)

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/AddUser/Confirm.cshtml [hunk 1]
```diff
@@ -5,24 +5,55 @@
...
-            <govuk-input asp-for="Email" type="email" disabled />
+            @if (!Model.HasCrmAccount)
+            {
+                <govuk-warning-text icon-fallback-text="Warning" data-testid="no-crm-account-warning">User does not have an account in CRM.<br />The user will not be able to access the TRS Console.</govuk-warning-text>
+            }
+            else if (Model.CrmAccountIsDisabled)
+            {
+                <govuk-warning-text icon-fallback-text="Warning" data-testid="disabled-crm-account-warning">User CRM account is disabled.<br />The user will not be able to access the TRS Console.</govuk-warning-text>
+            }
+            else if (Model.DqtRoles is null || Model.DqtRoles.Length == 0)
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/AddUser/Confirm.cshtml:25]:
```suggestion
            else if (Model.DqtRoles!.Length == 0)
```

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/AddUser/Confirm.cshtml [hunk 2]
```diff
@@ -5,24 +5,55 @@
...
+                        <h2 class="govuk-heading-m">Roles</h2>
+                    </govuk-checkboxes-fieldset-legend>
                     @foreach (var role in UserRoles.All)
                     {
                         <govuk-checkboxes-item value="@role">@UserRoles.GetDisplayNameForRole(role)</govuk-checkboxes-item>
                     }
                 </govuk-checkboxes-fieldset>
             </govuk-checkboxes>
 
+            @if (Model.DqtRoles is not null && Model.DqtRoles.Length > 0)
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/AddUser/Confirm.cshtml:46]:
```suggestion
            @if (Model.DqtRoles!.Length > 0)
```

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/EditUser.cshtml [hunk 3]
```diff
@@ -16,30 +16,48 @@
...
+                        <h2 class="govuk-heading-m">Roles</h2>
+                    </govuk-checkboxes-fieldset-legend>
                     @foreach (var role in UserRoles.All)
                     {
                         <govuk-checkboxes-item value="@role" disabled="@(!Model.IsActiveUser)">@UserRoles.GetDisplayNameForRole(role)</govuk-checkboxes-item>
                     }
                 </govuk-checkboxes-fieldset>
             </govuk-checkboxes>
 
+            @if (Model.DqtRoles is not null && Model.DqtRoles.Length > 0)
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/EditUser.cshtml:46]:
```suggestion
            @if (Model.DqtRoles?.Length > 0)
```

**Review by gunndabad — APPROVED** [review]

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/AddUser/Confirm.cshtml (hunk 2 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/AddUser/Confirm.cshtml]: Done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/EditUser.cshtml (hunk 3 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/EditUser.cshtml]: Done

**Pull request merged**